lib/child_process.js

```diff
@@ -841,17 +859,26 @@ function checkExecSyncError(ret, args, cmd) {
  *   }} [options]
  * @returns {Buffer | string}
  */
-function execFileSync(command, args, options) {
-  options = normalizeSpawnArguments(command, args, options);
+function execFileSync(file, args, options) {
+  const opts = normalizeExecFileArgs(file, args, options);
+
+  file = opts.file;
+  args = opts.args;
+  options = opts.options;
 
   const inheritStderr = !options.stdio;
-  const ret = spawnSync(options.file,
-                        ArrayPrototypeSlice(options.args, 1), options);
+  const ret = spawnSync(file, args, options);
 
   if (inheritStderr && ret.stderr)
     process.stderr.write(ret.stderr);
 
-  const err = checkExecSyncError(ret, options.args, undefined);
+  if (typeof options.argv0 === 'string') {
+    ArrayPrototypeUnshift(args, options.argv0);
+  } else {
+    ArrayPrototypeUnshift(args, file);
+  }
```

Comment:
Shouldn't this be part of `normalizeExecFileArgs()`?

Reply:
I prefer to move the validation to `normalizeExecFileArgs()` only. Because adding `options.argv0` to `args` is just to satisfy `checkExecSyncError()`